devtools/enforcer-rules/src/main/java/io/quarkus/enforcer/RequiresMinimalDeploymentDependency.java

```diff
@@ -0,0 +1,183 @@
+                .map(entry -> DEPLOYMENT_GAV_CACHE.computeIfAbsent(entry.getKey(), k -> parseDeploymentGAV(entry.getValue())))
+                .filter(optDeploymentGAV -> optDeploymentGAV
+                        .map(deploymentGAV -> !isMinDeploymentDepPresent(deploymentGAV, projArtifactKey, directDepsByGAV))
+                        .orElse(false))
+                .map(Optional::get)
+                .sorted()
+                .collect(Collectors.toList());
+
+        if (!missingDeploymentDeps.isEmpty()) {
+            String requiredDeps = missingDeploymentDeps.stream()
+                    .map(gav -> (Object[]) gav.split(":"))
+                    .map(gavArray -> String.format(DEP_TEMPLATE, gavArray))
+                    .collect(Collectors.joining("\n"));
+            throw new EnforcerRuleException(missingDeploymentDeps.size()
+                    + " *-deployment dependencies are missing/configured incorrectly:\n"
+                    + "    " + missingDeploymentDeps.stream().collect(Collectors.joining("\n    "))
+                    + "\n\nTo fix this issue, add the following dependencies to pom.xml:\n\n"
+                    + "        <!-- \"Symbolic\" test dependencies to *-deployment artifacts for consistent build order -->\n"
+                    + requiredDeps);
+        }
+    }
+
+    private String buildGAVKey(Artifact artifact) {
+        return artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + artifact.getVersion();
+    }
+
+    private Optional<String> parseDeploymentGAV(Artifact artifact) {
+        File artifactFile = artifact.getFile();
+        if (artifactFile == null || !artifactFile.exists()) {
+            throw new IllegalStateException("Artifact file not found for " + artifact);
+        }
+
+        Properties extProperties = new Properties();
+        try (ZipFile zipFile = new ZipFile(artifactFile)) {
+            ZipEntry entry = zipFile.getEntry(EXT_PROPERTIES_PATH);
+            if (entry == null) {
+                return Optional.empty();
+            }
+            extProperties.load(new InputStreamReader(zipFile.getInputStream(entry), StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed to read " + EXT_PROPERTIES_PATH + " from " + artifactFile, e);
+        }
+
+        String deploymentGAV = extProperties.getProperty("deployment-artifact");
+        if (deploymentGAV == null) {
+            throw new IllegalStateException(
+                    "deployment-artifact artifact not found in " + EXT_PROPERTIES_PATH + " from " + artifactFile);
+        }
+        return Optional.of(deploymentGAV);
+    }
+
+    private boolean isMinDeploymentDepPresent(String deploymentGAV, String projArtifactKey,
+            Map<String, Dependency> directDepsByGAV) {
+        return deploymentGAV.equals(projArtifactKey) // special case: current project itself is the "required dependency"
```

Comment:
This is a bit fishy, but I had to introduce this check for at least on it-submodule (forgot the name).
It almost feels like that submodule is violation the "no runtime deps in deployment modules" rule...

Reply:
I think the rule is opposite, "no deployment deps in runtime modules", and if there's a violation, then I guess we really want to know :-) Though I'm not sure if this indicates violation of that rule.

Reply:
"no deployment deps in runtime modules" makes sense as well but I meant: #11174

Reply:
Ah interesting, I didn't know about that one! Thanks.

Reply:
I just realized that #11174 is about runtime dependencies of _other_ extensions.
It is perfectly valid (and required!) that the deployment module of an extension A has a dependency on the runtime artifact of extension A.
It would not be valid if that deployment module would depend on the runtime artifact of an extension B!

Long story short: This check is not "fishy". I reworded #11174 a little for clarification.